code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -251,8 +251,10 @@ mob/living/simple_animal/hostile/hitby(atom/movable/AM as mob|obj,var/speed = TH
 		return FALSE
 
 	var/target_hit = FALSE
-
-	for(var/mob/M in check_trajectory(target_mob, src, pass_flags=PASSTABLE))
+	var/flags = PASSTABLE|PASSGLASS|PASSGRILLE
+	if(istype(projectiletype, /obj/item/projectile/beam))
+		flags &= ~(PASSGLASS|PASSGRILLE)

```

Comment:
The logic seems inverted here. Surely if it _is_ a beam, it _should have_ the extra pass flags? At the moment, it removes them.